src/helpers/radspec.js

```diff
@@ -48,53 +48,46 @@ export default (eth, evaluator) =>
     if (!fn) {
       // Even if we pass the ETH object, if it is not on mainnet it will use Aragon's ETH mainnet node
       // As the registry is the only available on mainnet
-      const registry = new MethodRegistry({ networkId: '1', eth })
-      const result = await registry.lookup(methodId)
+      const registry = new MethodRegistry({ networkId: '1', provider })
 
-      if (result) {
+      try {
+        const result = await registry.lookup(methodId)
         const { name } = registry.parse(result)
         return {
           type: 'string',
           value: name // TODO: should we decode and print the arguments as well?
         }
-      } else {
+      } catch {
         return makeUnknownFunctionNode(methodId)
       }
     }
     // If the function was found in local radspec registry. Decode and evaluate.
     const { source, sig } = fn
 
-    // get the array of input types from the function signature
-    const inputString = sig.replace(')', '').split('(')[1]
+    const fragment = ethers.utils.FunctionFragment.from(sig)
 
-    let parameters = []
+    const ethersInterface = new ethers.utils.Interface([fragment])
 
-    // If the function has parameters
-    if (inputString !== '') {
-      const inputs = inputString.split(',')
+    // Decode parameters
+    const args = ethersInterface.decodeFunctionData(fragment.name, data)
 
-      // Decode parameters
-      const parameterValues = ABI.decodeParameters(inputs, '0x' + data.substr(10))
-      parameters = inputs.reduce((acc, input, i) => (
-        {
-          [`$${i + 1}`]: {
-            type: input,
-            value: parameterValues[i]
-          },
-          ...acc
-        }), {})
-    }
+    const parameters = fragment.inputs.reduce(
+      (parameters, input, index) =>
+        Object.assign(parameters, {
```

Comment:
Ah, instead of the `Object.assign()`, we can use the object destructuring:

```
{
  [`$${index + 1}`]: {},
  ...parameters
}
```

For the most part, we've been moving away from `Object.assign()` as the other one is a bit more obvious (you can never accidentally mutate an existing object).